kcp/types.go

```diff
@@ -0,0 +1,14 @@
+package kcp
+
+// NamespaceReference holds a reference to a specific namespace within a cluster using KCP.
+type NamespaceReference struct {
+	// Namespace references a namespace within the cluster
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Namespace string `json:"namespace"`
+
+	// Workspace references a KCP workspace within the cluster
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])(":"[a-z0-9]([-a-z0-9]*[a-z0-9]))*$
```

Comment:
This needs to be updated I think. I can do it in my PR if you want. But we seem to need to be able to have more than one colon

Reply:
Here is an example workspace value `root:rh-sso-6340056:jbieren-q6jqm:jbieren-managed`

Reply:
That's actually right, but it seems that the colon doesn't need quotes. I fixed this: https://regex101.com/r/2ePsX7/1